arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,268 @@
+           let startingLocation = utilityNetwork.createElement(with: assetType, globalID: globalID),
+           let terminal = assetType.terminalConfiguration?.terminals.first(where: { $0.name == terminalName }) {
+            // Set the terminal for the location. (For our case, use the "Load" terminal.)
+            startingLocation.terminal = terminal
+            return startingLocation
+        }
+        return nil
+    }
+    
+    /// Get the utility tier's trace configuration.
+    func getTraceConfiguration() -> AGSUtilityTraceConfiguration? {
+        // Constants for creating the default trace configuration.
+        let domainNetworkName = "ElectricDistribution"
+        let tierName = "Medium Voltage Radial"
+        // Get a default trace configuration from a tier to update the UI.
+        let domainNetwork = utilityNetwork.definition.domainNetwork(withDomainNetworkName: domainNetworkName)
+        let utilityTierConfiguration = domainNetwork?.tier(withName: tierName)?.traceConfiguration
+        return utilityTierConfiguration
+    }
+    
+    /// Add a default phase to the list to better showcase the sample.
+    func addDefaultPhase() {
+        guard let defaultPhase = phaseChoices.first else { return }
+        phaseSummaries.append((defaultPhase, nil))
+        tableView.insertRows(at: [IndexPath(row: phaseSummaries.endIndex - 1, section: 0)], with: .automatic)
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func resetBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        phaseSummaries.removeAll()
+        tableView.reloadData()
+    }
+    
+    @IBAction func addBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        let selectedPhases = Set(phaseSummaries.map { $0.phase })
+        let remainingPhases = phaseChoices.filter { !selectedPhases.contains($0) }
+        guard !remainingPhases.isEmpty else { return }
+        let alertController = UIAlertController(
+            title: "Add a phase to get the load report.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        remainingPhases.forEach { phase in
+            alertController.addAction(UIAlertAction(title: phase.name, style: .default) { _ in
+                self.phaseSummaries.append((phase, nil))
+                self.tableView.insertRows(at: [IndexPath(row: self.phaseSummaries.endIndex - 1, section: 0)], with: .automatic)
+            })
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(cancelAction)
+        alertController.popoverPresentationController?.barButtonItem = sender
+        present(alertController, animated: true)
+    }
+    
+    @IBAction func runBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        guard !phaseSummaries.isEmpty,
+              let expression = initialExpression,
+              let phasesNetworkAttribute = phasesNetworkAttribute,
+              let traceParameters = traceParameters else { return }
```

Comment:
These should be non-nil, right? And if they are `nil`, the run button shouldn't be enabled. So could those properties be declared as implicitly unwrapped optionals?